### DISCUSSION TITLE
Add Recent Feed Tab

### DISCUSSION THREAD

**Discussion by jsonpreet** [deso-protocol/dips#16] [Q&A]
Please add the Recent Feed tab too, it's more helpful for all users to get the most recent posts, This feature available on all other Social Media Platforms. 

For now, in the Following Feed, I am getting old feeds, I have to refresh many times to get new posts. Sometimes I missed valuable posts from Creators.

I submitted PR related to this.  We can show Recent Feed Tab or In Left Sidebar with name Most Recent.

**Comment by tijno** [comment]
i think there are 2 separate things going on here:

1) following and global feeds do not update unless you do a hard browser refresh

this should be addressed, or atleast show a "3 posts have been posted - click to load"

2) admin recent posts feed shows all posts across the network

twitter & fb dont have a "all recent posts across all users" view

this is the change you are proposing.

It is not viable nor useful when Bitclout grows 2-3x let alone 100x

**Reply by jsonpreet** [reply]
I agreed with 1st

But Fb Have Recent Posts from Followers, We can change it with Recent Posts from Followers

**Reply by HPaulson** [reply]
Just to be clear -- the feed API endpoint has a specific Order field (forget exact naming). When you view the following feed, it IS "recent by following". It displays every post from your followings, in order. If it's missing something, it requires a hard refresh as @Tijn mentioned above.

If I'm reading everything correctly, what you're proposing already exists in the Following feed

**Reply by tijno** [reply]
> But Fb Have Recent Posts from Followers, We can change it with Recent Posts from Followers

they do and this was addresses by my point 1

but the change you proposed in the PR was to show a tab with all recent posts across entirety of bitclout for all users. 

As @hpaulson explained - the current following tab shows all recent posts by people you are following. 

It just doesnt refresh automatically when clicking around the site and can therefore become stale requiring hard reload

**Reply by bloated-devish** [reply]
Also, on a more general note, these aren't CIPs and probably should be discussed as issues IMO. 

CIPs should be much more focused dev discussions (with devs throwing fit on the naming just kidding).

Thanks \
Bloated

**Reply by HPaulson** [reply]
@bloated-devish Note that a Discussion doesn't become a CIP until approved to be a PR, thus not all discussions will CIPs, so I think this should be fine here. At least this is how I read readme

**Reply by jsonpreet** [reply]
@tijno ok i got all, can be there something for auto reload both feeds? Global and Following? Like after 3 or 5 seconds it reload and show new posts in feed.

I agreed with hard reload, but sometime have to reload 4 to 5 times then i get 2 or 3 posts, 

There is something devs have to improve in code 🙂

**Reply by tijno** [reply]
@harpreet9629 maybe best to close this discussion, and create a new one specifically focussed on that. 

I believe it would be a medium level change.

Reason - feeds previously did update, and admins removed that either by accident, but most likely on purpose to reduce load on their infrastructure.

**Comment by jsonpreet** [comment]
Sure no problem, my doubts are cleared.

Thank you @tijno

**Reply by 0xNordian** [reply]
@tijno 
> ...atleast show a "3 posts have been posted - click to load" 

This is a great idea.

**Reply by tijno** [reply]
It is but implementation much harder i suspect.
